cmd/vclusterctl/cmd/completion.go

```diff
@@ -117,3 +124,54 @@ func NewFishCommand() *cobra.Command {
 	}
 	return fishCmd
 }
+
+func listVClusterForCompletion(cmd *flags.GlobalFlags) ([]string, cobra.ShellCompDirective) {
+	if cmd.Context == "" {
+		rawConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{}).RawConfig()
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+
+		cmd.Context = rawConfig.CurrentContext
+	}
+
+	namespace := metav1.NamespaceAll
+	if cmd.Namespace != "" {
+		namespace = cmd.Namespace
+	}
+
+	vClusters, err := find.ListVClusters(cmd.Context, "", namespace)
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+	listClusters := []string{}
+	for _, v := range vClusters {
+		listClusters = append(listClusters, v.Name)
+	}
+	return listClusters, cobra.ShellCompDirectiveNoFileComp
+}
+
+func registerNamespaceCompletionFunc(rootCmd *cobra.Command) {
+	_ = rootCmd.RegisterFlagCompletionFunc("namespace", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		kubeClientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{})
+		restConfig, err := kubeClientConfig.ClientConfig()
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		client, err := kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
```

Comment:
This was the timeout being used for the [getPods](https://github.com/loft-sh/vcluster/blob/8f535b2647e578b09815633cacc2cd26f34acb76/cmd/vclusterctl/cmd/find/find.go#L207) function so I used the same. But I see that `infracloud/main` repo the timeout is different.

Reply:
Now that you have pointed it out I feel that in case of auto-completion it should be lesser because otherwise it will keep the command line (or user) blocked in case the server times out. So I am planning to change it to 3 sec. Does that looks like a good timeout?